### PR TITLE
Fix Complete->Pending transition to alter stats and logs 

### PR DESCRIPTION
**Pull request by cklosowski**
#4052

### CODE REVIEW

File: includes/payments/class-edd-payment.php [hunk 1]
```diff
@@ -1568,11 +1569,81 @@ private function process_refund() {
...
+		}
+
+		// Allow extensions to filter for their own payment types, Example: Recurring Payments
+		$process_pending = apply_filters( 'edd_should_process_pending', $process_pending, $this );
+
+		if ( false === $process_pending ) {
+			return;
+		}
+
+		do_action( 'edd_pre_pending_payment', $this );
```
Comment (pippinsplugins) [inline includes/payments/class-edd-payment.php]: I'm not sure this action name makes sense unless this method also fires when a pending payment is initially created.

Comment (cklosowski) [inline includes/payments/class-edd-payment.php]: It does not, any suggestions...something like `edd_pre_to_pending_payment`?

Comment (pippinsplugins) [inline includes/payments/class-edd-payment.php]: How about no actions aside from the status status change ones?

Comment (cklosowski) [inline includes/payments/class-edd-payment.php]: Works for me.

**Comment by cklosowski** [comment]
@pippinsplugins take a look at that?

**Comment by pippinsplugins** [comment]
Looks good to me. Get some PHPDocs on there and we should be good to merge.

**Comment by cklosowski** [comment]
sounds good!

**Comment by cklosowski** [comment]
Merging since PHPdocs are updated.

**Pull request merged**